topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeAppliers.java

```diff
@@ -29,9 +30,10 @@ interface OperationApplier {
      * operation. For example, an operation for joining a partition can mark the state as JOINING.
      *
      * @return an either which contains an exception if the operation is not valid, or a function to
-     *     update the cluster toplogy
+     *     update the cluster topology
      */
-    Either<Exception, UnaryOperator<MemberState>> init();
+    Either<Exception, UnaryOperator<MemberState>> init(
+        final ClusterTopology currentClusterTopology);
```

Comment:
:thought_balloon: Since we provide current cluster topology as parameter, we could also return the updated topology instead of the UnaryOperator. However, returning the operator may be future-proof in case we want to allow applying operations in parallel.

Reply:
Yep, I like it this way because it makes clear that init is only allowed to update a single member state, not the entire topology.